website/client/components/shops/buyModal.vue

```diff
@@ -402,13 +402,8 @@
           return;
         }
 
-        if (this.item.currency === 'gems' &&
-          !confirm(this.$t('purchaseFor', { cost: this.item.value * this.selectedAmountToBuy }))) {
-          return;
-        }
-
-        if (this.item.currency === 'hourglasses' &&
-          !confirm(this.$t('purchaseForHourglasses', { cost: this.item.value }))) {
+        const shouldConfirmPurchase = this.item.currency === 'gems' || this.item.currency === 'hourglasses';
+        if (shouldConfirmPurchase && !this.confirmPurchase(this.item.currency, this.item.value * this.selectedAmountToBuy)) {
```

Comment:
If it is desirable to ask for confirmation for gold-purchasable items as well(?) I could update this line to be:
```suggestion
        if (!this.confirmPurchase(this.item.currency, this.item.value * this.selectedAmountToBuy)) {
```

Reply:
For most gold-purchasable items, I'd say that there should not be a confirmation step because it's a bit annoying and because losing small amounts of gold accidentally isn't a big issue. Avoiding the annoyance of the extra step is more important.

Gold-purchasable quest scrolls are an exception because they cost a LOT of gold (so accidental purchases can be very upsetting to users who don't know about [FCV](http://habitica.fandom.com/wiki/Fix_Character_Values)), and they are bought rarely so the annoyance of the confirmation step isn't a big factor.

Reply:
Makes sense to me. I'll leave the proposed change as-is then.